axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveServiceImpl.java

```diff
@@ -438,9 +439,25 @@ public String realize(StockMove stockMove) throws AxelorException {
   }
 
   @Override
-  @Transactional(rollbackOn = {Exception.class})
   public String realize(StockMove stockMove, boolean checkOngoingInventoryFlag)
       throws AxelorException {
+    String newStockSeq = realizeStockMove(stockMove, checkOngoingInventoryFlag);
+
+    if (stockMove.getTypeSelect() == StockMoveRepository.TYPE_INCOMING
+        && !stockMove.getIsReversion()) {
+      partnerProductQualityRatingService.calculate(stockMove);
```

Comment:
this computation should be inside the transaction, please make this call inside `realizeStockMove` method